### DISCUSSION TITLE
vite-plugin-solid - doesn't seem to work for dependencies containing JSX

### DISCUSSION THREAD

**Discussion by bmingles** [solidjs/solid#2346] [Q&A] [answered]
I'm using the standard project typescript starter (`npx degit solidjs/templates/ts my-app`). I've installed a dependency which contains JSX components. When I npm start the project, I get `React is not defined` errors. It looks like the JSX in the dependency is getting output as `React.createElement` instead of the SolidJS equivalent. I would have expected `vite-plugin-solid` to somehow handle this, but seems like it does not. Any suggestions on how to address this?

**Comment by atk** [comment, marked as answer]
Vite-plugin-solid does not magically change react JSX components into solid JSX components. Your expectation is the issue here.

**Reply by bmingles** [reply]
Thanks. I now realize I need a build step as part of my npm package deployment and can't serve raw JSX from the lib package.

**Reply by bmingles** [reply]
Related to this request https://github.com/solidjs/vite-plugin-solid/issues/97

**Reply by atk** [reply]
If you want to serve raw jsx, you need to export it under a "solid" condition.

**Reply by bmingles** [reply]
@atk is there an example of what such an "solid" condition would look like? I currently have:

```json
"exports": {
  "solid": {
    "development": "./dist/index.js",
    "import": "./dist/index.js"
  }
}
```

Where the index.js file imports solid JSX components.
